Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -2456,6 +2458,49 @@ void LocalDOMWindow::finishedLoading()
     }
 }
 
+std::optional<size_t> LocalDOMWindow::initializeEventTimingEntry(const Event& event, EventType type)
+{
+    if (!event.isTrusted() || !EventCounts::IsCounted(type))
+        return std::nullopt;
+
+    LOG_WITH_STREAM(PerformanceTimeline, stream << "Initializing event timing entry of type " << event.type());
+
+    // event.timeStamp() may not be reliable (wall clock):
+    auto startTime = std::min(event.timeStamp(), MonotonicTime::now());
+    m_performanceEventTimingCandidates.push_back(PerformanceEventTimingCandidate {
```

Comment:
This will slow down every event dispatch. We need to find an alternative design which doesn't involve slowing down every event dispatch. We shouldn't be creating `PerformanceEventTimingCandidate` at all in most cases given the minimum threshold is 16ms.